### PR TITLE
Basic Eio wrapper

### PR DESCRIPTION
**Pull request by Willenbrink**
This PR ports the Lwt wrapper to Eio, thereby providing a nicer interface for those using Eio. The code structure is still quite similar, it could likely be written more compactly.

I've also written a test and used it a bit so the code is at least mostly correct. I have not tested it extensively so an infinite wait is in theory possible.

Unrelated changes in the PR:
- In the dune-project file `(implicit_transitive_deps false)` caused some issues and after finding that it is an experimental feature I removed it. Is there a reason for its existence?
- I changed the "executables" in `test/` into actual tests in the dune file. Is there a reason this hasn't been done or is it legacy?

### CODE REVIEW

File: multipart_form-lwt.opam [hunk 1]
```diff
@@ -17,7 +17,7 @@ depends: [
   "bigstringaf"
   "ke"
   "lwt"
-  "multipart_form" {= version}
```
Comment (dinosaure) [inline multipart_form-lwt.opam:20]: Why you delete this constraint?

File: test/dune
Comment (dinosaure) [inline test/dune:1]: `(rule ...` and `(test ...` usage is a legacy one. It would be more interesting to make this change in another PR. Finally, from my point of view, it is not fundamentally useful to update this part.

File: lib_eio/multipart_form_eio.mli [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+  identify:(Header.t -> 'id) ->
+  string Eio.Stream.t ->
+  Content_type.t ->
+  ('id t, [> `Msg of string ]) result Eio.Promise.t
+  * ('id * Header.t * string Eio.Stream.t) Eio.Stream.t
+(** [stream ~identify src content_type] returns:
+    - a promise [th] about the parser
+    - a stream [stream] of parts
+
+    They can be manipulated with [Lwt.both], and, by this way,
```
Comment (dinosaure) [inline lib_eio/multipart_form_eio.mli:17]: You probably should update the documentation as well and describe how to use this function with `eio`.

File: multipart_form-lwt.opam (hunk 1 above)
Comment (Willenbrink) [reply multipart_form-lwt.opam:20]: I've readded the constraint now. opam complained otherwise but I just noticed now that opam states version `0.4.1` for both multipart_form and multipart_form-lwt but `~dev` for the eio package. I'm not familiar with opam packaging, is this pulled from tags in the git repository?

File: test/dune
Comment (Willenbrink) [reply test/dune:1]: I've reverted the change, the test for eio follows your example now.

File: lib_eio/multipart_form_eio.mli (hunk 2 above)
Comment (Willenbrink) [reply lib_eio/multipart_form_eio.mli]: The documentation has been updated now. Thanks for your feedback!

**Comment by dinosaure** [comment]
The CI still fails :confused:, can you fix it?

**Comment by dinosaure** [comment]
> I'm not sure how to resolve the other failures. They are all on compiler version < 5.0 so obviously they cannot build with Eio. I guess it's not a problem for multipart_form-eio but it also affects the tests I added. Should I simply remove them or how do you want to proceed?

Currently, your tests are just a copy to what it exists and what is really matter is if `multipart_form` works really (independantly than `lwt` or `eio` or something else). If removed `eio` tests helps to have a green CI, it is certainly the best option.

**Comment by Willenbrink** [comment]
All the tests pass now! Well, except the experimental lint-lower-bounds but I guess that is expected.

**Comment by dinosaure** [comment]
Thanks, I will try to cut a release as soon as I can.
